yamllint/rules/quoted_strings.py

```diff
@@ -235,7 +274,8 @@ def check(conf, token, prev, next, nextnext, context):
                     quote_type)
 
         # But when used need to match config
-        elif token.style and not _quote_match(quote_type, token.style):
+        elif token.style and not is_quoted_quote and not _quote_match(
+                quote_type, token.style):
```

Comment:
For better performance (because yamllint can take some time to run on large repositories...), I propose to compute "*does this string contain quoted quote?*" only if needed. This means computing this in last resort (after `_quote_match()`).

Something like:
```suggestion
        elif (token.style and not _quote_match(quote_type, token.style) and
                not (conf['skip-quoted-quote'] and _has_quoted_quotes(token))):
```

Reply:
I extracted `is_quoted_quote` variable logic to `_has_quoted_quotes` function and added `conf['allow-quoted-quotes']` check in `if` clauses. Also, I rearranged `if` statements and placed quoted quotes things to the end.

Reply:
Perfect.